docs/js/features/odata/generate-client.md

```diff
@@ -0,0 +1,115 @@
+---
+id: generator-js-sdk
+title: Generate a typed OData client for JavaScript
+hide_title: false
+hide_table_of_contents: false
+sidebar_label: Generate an OData client for JavaScript
+keywords:
+- sap
+- cloud
+- sdk
+- odata
+- JavaScript 
+- TypeScript
+- generate
+author: Charles Dubois
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+## Generate a typed OData client with the OData Generator
+
+The generator allows you to generate a custom OData client library for any service. You can then easily access these services via the client libraries.
+
+There are two way to use the generator:
+- Via the command line interface (CLI)
+- By instantiating and invoking it at runtime
+
+For any of them you need to have an `EDMX` file holding the service metadata.
+
+## Installation
+
+Run this command in your project's terminal to install the generator globally.
+```sh
+npm install -g @sap-cloud-sdk/generator
+```
+
+:::note
+The generator is included in the SAP cloud-sdk-cli, if you have it installed you don't need to install the generator.
+:::
+
+## Usage
+
+### Using the CLI
+
+The generator is primarily meant to be used on the command line:
+
+```sh
+generate-odata-client --inputDir path/to/your/service-specification(s) --outputDir path/where/the/modules/are/stored
+```
+
+Run `generate-odata-client --help` for further options.
+
+### Invoke the generator programmatically
+
+You can also use the generator programmatically. You will have to provide the options anyways.
+
+1. Import the generator in your project.
+```ts
+import { generateProject } from '@sap-cloud-sdk/generator';
+```
+
+2. Create your options, adapt the input & output directory as well as the package name according to your setup.
+```ts
+const options: GeneratorOptions = initializeOptions();
+```
+3. Create your project datastructure with all sourcefiles based on your options
+```ts
+const project = generateProject(options);
+```
+
+4. Modify your project if you need to, then save the files at the specified location.
+```ts
+project.save();
+```
+3. & 4. Alternatively you can generate and save the project in one step with:
+```ts
+generate(options)
+```
+
+## Options
+
+:::note
+All parameters can be used in the command line, even if they have no alias:  
```

Comment:
Why would that not be the case if there is no alias?! I would move that section up to the CLI part. It makes more sense there.